New default `geb` feature implementation

matrei opened the pull request:
- Rename feature `geb` to `geb-with-webdriver-binaries` and make it an optional feature
- Add new default feature `geb-with-testcontainers`

matrei committed:
feat: New default `geb` feature implementation

- Rename feature `geb` to `geb-with-webdriver-binaries` and make it an optional feature
- Add new default feature `geb-with-testcontainers`

matrei committed:
test: Double the test timeout

Because tests are timing out

matrei committed:
Revert "test: Double the test timeout"

This reverts commit f72c6690a379ee816f2bdd94739c422d7d81b230.

matrei committed:
test: Make tests compatible with Windows

sbglasius reviewed (approved):
Looks good

matrei commented:
The build is failing right now when running tests.
This is mimicking what the test is trying to do:
```console
mattias@mattias-z4-linux:/tmp/grailsforgetmp15664837821511079914$ ./gradlew build

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':buildSrc:buildScriptClasspath'.
> Could not resolve org.apache.groovy:groovy:4.0.24-SNAPSHOT.
  Required by:
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118 > org.apache.groovy:groovy-ant:4.0.23
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118 > org.apache.groovy:groovy-json:4.0.23
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118 > org.apache.groovy:groovy-jmx:4.0.23
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118 > org.apache.groovy:groovy-xml:4.0.23
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118 > org.apache.groovy:groovy-bom:4.0.23
   > Could not resolve org.apache.groovy:groovy:4.0.24-SNAPSHOT.
      > Could not get resource 'https://repo.grails.org/grails/core/org/apache/groovy/groovy/4.0.24-SNAPSHOT/groovy-4.0.24-SNAPSHOT.pom'.
         > Could not GET 'https://repo.grails.org/grails/core/org/apache/groovy/groovy/4.0.24-SNAPSHOT/groovy-4.0.24-SNAPSHOT.pom'.
            > Read timed out
> Could not resolve org.grails:grails-datastore-core:9.0.0-SNAPSHOT.
  Required by:
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-web:9.0.0-SNAPSHOT:20241025.200433-19
   > Skipped due to earlier error
> Could not resolve org.grails:grails-datastore-gorm:9.0.0-SNAPSHOT.
  Required by:
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-gorm-support:9.0.0-SNAPSHOT:20241025.200433-19
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-gorm-hibernate5:9.0.0-SNAPSHOT:20241026.013223-23
   > Skipped due to earlier error
> Could not resolve org.grails:grails-core:7.0.0-SNAPSHOT.
  Required by:
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-gorm-support:9.0.0-SNAPSHOT:20241025.200433-19
   > Skipped due to earlier error
> Could not resolve org.apache.groovy:groovy:4.0.24-SNAPSHOT.
  Required by:
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-web:9.0.0-SNAPSHOT:20241025.200433-19
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-gorm-support:9.0.0-SNAPSHOT:20241025.200433-19
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-gorm-hibernate5:9.0.0-SNAPSHOT:20241026.013223-23
   > Could not resolve org.apache.groovy:groovy:4.0.24-SNAPSHOT.
      > Could not get resource 'https://repo.grails.org/grails/core/org/apache/groovy/groovy/4.0.24-SNAPSHOT/groovy-4.0.24-SNAPSHOT.pom'.
         > Could not GET 'https://repo.grails.org/grails/core/org/apache/groovy/groovy/4.0.24-SNAPSHOT/groovy-4.0.24-SNAPSHOT.pom'.
            > Read timed out

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 24s
1 actionable task: 1 up-to-date
```

matrei committed:
chore: Update license header of changed file

jamesfredley commented on grails-cli/src/test/groovy/org/grails/forge/cli/CommandSpec.groovy:
```diff
@@ -47,7 +47,7 @@ class CommandSpec extends Specification {
         StringBuilder gradleCommand = new StringBuilder()
         if (spock.util.environment.OperatingSystem.current.isWindows()) {
-            gradleCommand.append("gradlew.bat")
+            gradleCommand.append("cmd.exe /c gradlew")
```
Thank you for finding this solution!

jamesfredley commented:
geb-with-testcontainers should allow Gradle to go back to 8.10.2.

geb-with-webdriver-binaries will be stuck at 8.6, for now.

matrei committed:
test: Re-enable integration tests

After switching to `ContainerGebSpec`, it is no longer necessary to have the integration tests disabled.

Related: #385

matrei committed:
chore: Cleanup changed files

matrei commented:
> geb-with-testcontainers should allow Gradle to go back to 8.10.2.
> 
> geb-with-webdriver-binaries will be stuck at 8.6, for now.

@jamesfredley Is it possible to differentiate the gradle version based on selected features?
Do you think we should update to Gradle 8.10.2 even though the optional `geb-with-webdriver-binaries` won't work without manually downgrading created apps?
